patches/net/minecraft/client/renderer/entity/EntityRenderer.java.patch

```diff
@@ -9,6 +9,14 @@
              this.renderNameTag(p_364816_, p_364816_.nameTag, p_114488_, p_114489_, p_114490_);
          }
      }
+@@ -245,6 +_,7 @@
+     public final S createRenderState(T p_361382_, float p_360885_) {
+         S s = this.reusedState;
+         this.extractRenderState(p_361382_, s, p_360885_);
++        net.neoforged.neoforge.client.renderstate.RenderStateExtensions.onUpdateEntityRenderState(this, p_361382_, s);
```

Comment:
I've worked with this system for a bit in a mod of mine that renders health bars above mobs and therefore must add the relevant information to the render state. 

From that experience I feel like it would be more useful here to instead fire a new all purpose event for modifying the created render state, offering the ability to both modify existing render state values set by vanilla, and for manually adding new values via `IRenderStateExtension`.

This new event could exist in addition to `RegisterRenderStateModifiersEvent`, but could also outright replace it, since I'm not sure it's really necessary here to take the approach of registering a bunch of filters and consumers. Just letting modders handle things when the render state is actually created should work fine.

The new event could also be a potential replacement for `RenderNameTagEvent$CanRender` as it would also allow for overriding the name tag set to the render state.

Reply:
You can still modify vanilla values with this system, its just not guaranteed to stay that same value. In fact, even with an event it wouldn't be guaranteed. Additionally, the idea of firing another event every render frame for every entity was shut down repeatedly for performance concerns, which is why you see this implementation instead.